core/src/main/java/org/elasticsearch/action/ingest/SimulateExecutionService.java

```diff
@@ -38,20 +39,38 @@
         this.threadPool = threadPool;
     }
 
+    void executeVerboseDocument(Processor processor, IngestDocument ingestDocument, List<SimulateProcessorResult> processorResultList) throws Exception {
```

Comment:
yup! will add unit tests for all the branching that can occur within this method